e2e/credentials-server.e2e.spec.ts

```diff
@@ -0,0 +1,197 @@
+
+    if (body.length) {
+      args.push('--data', body);
+    }
+    const { stdout, stderr } = await spawnFile(command, args, { stdio: 'pipe' });
+
+    expect(stderr).toEqual('');
+
+    return stdout;
+  }
+
+  async function doRequestExpectStatus(path: string, body: string, expectedStatus: number) {
+    const args = initialArgs.concat(['-v', `http://${ ipaddr }:${ serverState.port }/${ path }`]);
+
+    if (body.length) {
+      args.push('--data', body);
+    }
+    const { stderr } = await spawnFile(command, args, { stdio: 'pipe' });
+
+    expect(stderr).toContain(`HTTP/1.1 ${ expectedStatus }`);
+  }
+
+  test.describe.configure({ mode: 'serial' });
+
+  test.beforeAll(async() => {
+    createDefaultSettings();
+    electronApp = await _electron.launch({
+      args: [
+        appPath,
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ]
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({
+      screenshots: true,
+      snapshots:   true
+    });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: playwrightReportAssets(path.basename(__filename)) });
+    await electronApp.close();
+  });
+
+  test('should emit connection information', async() => {
+    const dataPath = path.join(paths.appHome, 'credential-server.json');
+    const dataRaw = await fs.promises.readFile(dataPath, 'utf-8');
+
+    serverState = JSON.parse(dataRaw);
+    expect(typeof serverState.user).toBe('string');
+    expect(typeof serverState.password).toBe('string');
+    expect(typeof serverState.port).toBe('number');
+    expect(typeof serverState.pid).toBe('number');
+
+    // Now is a good time to initialize the various connection-related values.
+    authString = `${ serverState.user }:${ serverState.password }`;
+    if (os.platform() === 'win32') {
+      ipaddr = wslHostIPv4Address();
+      if (!ipaddr) {
+        throw new Error('Failed to get the WSL IP address');
+      }
+      // arguments for wsl
+      initialArgs = ['--distribution', 'rancher-desktop', '--exec', 'curl'];
+    } else {
+      ipaddr = 'localhost';
+    }
+    // common arguments for curl
+    initialArgs = initialArgs.concat(['--silent', '--user', authString, '--request', 'POST']);
+  });
+
+  test('should require authentication', async() => {
+    const url = `http://${ ipaddr }:${ serverState.port }/list`;
+    const resp = await fetch(url);
+
+    expect(resp.ok).toBeFalsy();
+    expect(resp.status).toEqual(401);
+  });
+
+  test('should be able to use the API', async() => {
+    const bobsURL = 'https://bobs.fish/tackle';
+    const bobsFirstSecret = 'loblaw';
+    const bobsSecondSecret = 'shoppers with spaces and % and \' and &s and even a 😱';
```

Comment:
I think I saw errors with the emoji on Windows, but will need to confirm.

Reply:
I didn't see errors, and it looks fine in both notepad and wsl/vim.